core-tests/src/test/java/org/mule/el/mvel/datatype/SessionVarExpressionDataTypeResolverTestCase.java

```diff
@@ -7,13 +7,87 @@
+import org.mule.el.mvel.StaticVariableResolverFactory;
+import org.mule.el.mvel.VariableVariableResolverFactory;
+import org.mule.mvel2.MVEL;
+import org.mule.mvel2.ParserConfiguration;
+import org.mule.mvel2.ParserContext;
+import org.mule.mvel2.compiler.CompiledExpression;
+import org.mule.mvel2.integration.impl.CachedMapVariableResolverFactory;
+import org.mule.tck.junit4.AbstractMuleContextTestCase;
+import org.mule.transformer.types.DataTypeFactory;
 
-public class SessionVarExpressionDataTypeResolverTestCase extends AbstractVarExpressionDataTypeResolverTestCase
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+
+import org.junit.Test;
+
+public class SessionVarExpressionDataTypeResolverTestCase extends AbstractMuleContextTestCase
 {
+    public static final String EXPRESSION_VALUE = "bar";
+    public static final String CUSTOM_ENCODING = StandardCharsets.UTF_16.name();
+    public static final String PROPERTY_NAME = "foo";
+
+    private final ExpressionDataTypeResolver expressionDataTypeResolver = new SessionVarExpressionDataTypeResolver();
+    private final String variableName = "sessionVars";
+
+    @Test
+    public void returnsFlowVarDataTypeUsingMapSyntax() throws Exception
+    {
+        doVarDataTypeTest(variableName + "['" + PROPERTY_NAME + "']");
+    }
+
+    @Test
+    public void returnsFlowVarDataTypeUsingDotSyntax() throws Exception
+    {
+        doVarDataTypeTest(variableName + "." + PROPERTY_NAME );
+    }
+
+    @Test
+    public void returnsFlowVarDataTypeUsingEscapedDotSyntax() throws Exception
+    {
+        doVarDataTypeTest(variableName + ".'" + PROPERTY_NAME + "'");
+    }
+
+    protected void doVarDataTypeTest(String expression) throws Exception
+    {
+        final DataType expectedDataType = DataTypeFactory.create(String.class, JSON);
+        expectedDataType.setEncoding(CUSTOM_ENCODING);
+
+        MuleEvent testEvent = getTestEvent(TEST_MESSAGE);
+        testEvent.getSession().setProperty(PROPERTY_NAME, EXPRESSION_VALUE, expectedDataType);
 
-    public SessionVarExpressionDataTypeResolverTestCase()
+        final ParserConfiguration parserConfiguration = MVELExpressionLanguage.createParserConfiguration(Collections.EMPTY_MAP);
+        final MVELExpressionLanguageContext context = createMvelExpressionLanguageContext(testEvent, parserConfiguration);
+
+        CompiledExpression compiledExpression = (CompiledExpression) compileExpression(expression, new ParserContext(parserConfiguration));
+        // Expression must be executed, otherwise the variable accessor is not properly configured
+        MVEL.executeExpression(compiledExpression, context);
+
+        assertThat(expressionDataTypeResolver.resolve(testEvent, compiledExpression), like(String.class, JSON, CUSTOM_ENCODING));
+    }
+
+    protected MVELExpressionLanguageContext createMvelExpressionLanguageContext(MuleEvent testEvent, ParserConfiguration parserConfiguration)
```

Comment:
This method and the test look like a duplicate of FlowVarExpressionDataTypeResolverTestCase